[Linux] Publish Specialization complete event from functions host

balag0 opened the pull request:
This event will be used by the init process in the container to track functions runtime status. A delay of 5 seconds is added to avoid impacting cold start.

<!-- Please provide all the information below.  -->

https://github.com/Azure/azure-functions-docker-private/issues/136

resolves https://github.com/Azure/azure-functions-docker-private/issues/136

### Pull request checklist

* [X ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ X] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR https://github.com/Azure/azure-functions-docker-private/issues/136
* [ X] I have added all required tests (Unit tests, E2E tests)

<!-- Optional: delete if not applicable  -->
### Additional information

Additional PR information

pragnagopa commented on src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs:
```diff
@@ -16,23 +16,27 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
 {
     public class LinuxContainerActivityPublisher : IHostedService, IDisposable, ILinuxContainerActivityPublisher
     {
+        public const string SpecializationCompleteEvent = "Specialization complete";
```
nit: move this to `ScriptContants` and update the value to be more descriptive:
`public const string LinuxConsumptionSpecializationCompletedEvent = "Specialized";` ?

balag0 replied:
Yes, I am thinking this is not needed outside this class (other than the test) so leaning towards leaving it here for now and move it to global constants if things change later?

pragnagopa replied:
Ok to leave it here. I recommend changing it to `Specialized` or `SpecializationCompleted` - without the space

balag0 replied:
Yes. Fixed in the current iteration.

pragnagopa commented on src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs:
```diff
@@ -96,8 +102,30 @@ public Task StopAsync(CancellationToken cancellationToken)
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"{nameof(PublishSpecializationCompleteEvent)} failed with {e}");
```
nit: use parameterized logging 
` _logger.LogWarning("{PublishSpecializationCompleteEvent} failed with {ex}", nameof(PublishSpecializationCompleteEvent), ex);`

balag0 replied:
Fixed.

pragnagopa reviewed (commented):
Added minor comments. Thanks!

balag0 committed:
[Linux Consumption] Publish Specialization complete event from functions host

This event will be used by the init process in the container to track container health. A delay of 5 seconds is added to avoid impacting cold start.
